src/table/sticky-columns/use-sticky-columns.ts

```diff
@@ -208,18 +210,16 @@ export function useStickyCellStyles({
         }
       };
 
-      const unsubscribe = stickyColumns.store.subscribe(selector, (newState, prevState) =>
-        updateCellStyles(selector(newState), selector(prevState))
-      );
-      return () => {
-        unsubscribe();
-        // Force the cleanup
-        updateCellStyles(null, { padLeft: false, lastLeft: false, lastRight: false, offset: {} });
-      };
+      // If the node is not null (i.e., the table cell is being mounted or updated, not unmounted),
+      // set up a new subscription to the store's updates
+      if (node) {
+        unsubscribeRef.current = stickyColumns.store.subscribe(selector, (newState, prevState) => {
+          updateCellStyles(selector(newState), selector(prevState));
+        });
+      }
     },
-    // getClassName is expected to be pure
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [stickyColumns.store, stickyColumns.isEnabled, columnId]
+
+    [columnId, setCell, stickyColumns.store, getClassName]
```

Comment:
I think we should probably assume the getClassName is pure and exclude it from the dependency array. Otherwise it is going to have an impact on performance unless the getClassName is memoized by the consumer